src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs

```diff
@@ -209,7 +208,7 @@ private unsafe void FindNextEntry()
 
             // We need more data
             if (GetData())
-                _entry = (Interop.NtDll.FILE_FULL_DIR_INFORMATION*)_pinnedBuffer.AddrOfPinnedObject();
+                _entry = (Interop.NtDll.FILE_FULL_DIR_INFORMATION*)_buffer.ToPointer();
```

Comment:
Nit: is the `.ToPointer()` actually needed?

Reply:
I'll double check. I think we might have to cast to `void*` first otherwise?